tensorflow/python/saved_model/load_test.py

```diff
@@ -2547,6 +2549,63 @@ def __call__(self, inputs):
     self.assertAllEqual(imported(constant_op.constant(["d", "b"])),
                         [3, 1])
 
+class _TestModel(module.Module):
+
+  def __init__(self, rows, cols):
+    super().__init__()
+    self.rows=rows
+    self.cols=cols
+    self.table = None
+
+  def __call__(self, x):
+    with ops.device('/cpu:0'):
+      self.table = variables.Variable(
+          constant_op.constant(1., shape=[self.rows, self.cols]))
+      x = math_ops.matmul(self.table, x)
+      x = math_ops.reduce_sum(x, axis=0)
+    return x
+
+class SavedModelLoadMemoryTests(test.TestCase):
+
+  def get_gpu_memory(self):
+    command = "nvidia-smi --query-gpu=memory.total --format=csv"
+    mem_free_info = (
+        subprocess.check_output(command.split())
+        .decode('ascii')
+        .split('\n')[:-1][1:]
+    )
+    mem_free_values = [int(x.split()[0]) for i, x in enumerate(mem_free_info)]
+    if isinstance(mem_free_values, list):
+      return int(mem_free_values[0] / 1024)
+    return int(mem_free_values)
+
+  @test_util.run_gpu_only
+  def test_no_oom_loading_large_tenor(self):
+    if not config.get_soft_device_placement():
+      self.skipTest("This test only works for soft device placement is on")
+    max_gpu_memory_in_Gb = self.get_gpu_memory()
+    save_dir = os.path.join(self.get_temp_dir(), "saved_model")
+    ncols = 128
+    nrows = (max_gpu_memory_in_Gb + 1) * 2**30 // ncols // 4
+    model = _TestModel(rows=nrows, cols=ncols)
+    x = array_ops.zeros(shape=(ncols, 128), dtype=dtypes.float32)
+    y = model(x)
+    save.save(
+        model,
+        save_dir,
+        options=save_options.SaveOptions(
+            experimental_variable_policy=save_options.VariablePolicy
+                .SAVE_VARIABLE_DEVICES
+        ),
+    )
+    loaded = load.load(
```

Comment:
A test that runs subprocess raises flags in my head. I would drop the OOM part of the test, and instead validate that this variable has been placed on the GPU as expected.